README.md

```diff
@@ -21,135 +21,171 @@ gem 'acme-client'
 
-# You may need to agree to the terms of service (that's up the to the server to require it or not but boulder does by default)
-registration.agree_terms
+You can access the challenge you wish to complete using the `#dns` or `#http` method.
+
+```ruby
+order = client.new_order(identifiers: ['example.com'])
+authorization = order.authorizations.first
+challenge = authorization.http
 ```
 
-### Authorize for domain
+### Preparing for HTTP challenge
+
+To complete the HTTP challenge, you must prepare return a file using HTTP.
 
-Before you are able to obtain certificates for your domain, you have to prove that you are in control of it.
+The path follow the following format:
+
+> .well-known/acme-challenge/#{token}
+
+And the file content is the key authorization. The HTTP01 object as utility methods to generate them.
 
 ```ruby
-authorization = client.authorize(domain: 'example.org')
+> http_challenge.content_type # => 'text/plain'
+> http_challenge.file_content # => example_token.TO1xJ0UDgfQ8WY5zT3txynup87UU3PhcDEIcuPyw4QU
+> http_challenge.filename # => '.well-known/acme-challenge/example_token'
+> http_challenge.token # => 'example_token'
+```
 
-# If authorization.status returns 'valid' here you can already get a certificate
-# and _must not_ try to solve another challenge.
-authorization.status # => 'pending'
+For test purpose you can just save the challenge file and use Ruby to serve it:
 
-# You can can store the authorization's URI to fully recover it and
-# any associated challenges via Acme::Client#fetch_authorization.
-authorization.uri # => '...'
+```bash
+ruby -run -e httpd public -p 8080 --bind-address 0.0.0.0
+```
 
-# This example is using the http-01 challenge type. Other challenges are dns-01 or tls-sni-01.
-challenge = authorization.http01
+### Preparing for DNS challenge
 
-# The http-01 method will require you to respond to a HTTP request.
+To complete the DNS challenge, you must set a DNS record to prove that you control the domain.
 
-# You can retrieve the challenge token
-challenge.token # => "some_token"
+The DNS01 object as utility methods to generate them.
 
-# You can retrieve the expected path for the file.
-challenge.filename # => ".well-known/acme-challenge/:some_token"
+```ruby
+> dns_challenge.record_name # => '_acme-challenge'
+> dns_challenge.record_type # => 'TXT'
+> dns_challenge.record_content # => 'HRV3PS5sRDyV-ous4HJk4z24s5JjmUTjcCaUjFt28-8'
+```
 
-# You can generate the body of the expected response.
-challenge.file_content # => 'string token and JWK thumbprint'
+### Request a challenge verification
 
-# You are not required to send a Content-Type. This method will return the right Content-Type should you decide to include one.
-challenge.content_type
+Once your are ready to complete the challenge, you can request to server to perform the verification.
```

Comment:
"your are" to "you are"

Perhaps "you can request the server perform the verification"